pkg/storagepool/storagepool.go

```diff
@@ -162,3 +165,97 @@ func (v *casPoolOperation) Delete() (*v1alpha1.CasPool, error) {
 	}
 	return pool, nil
 }
+
+type storagePoolOperation struct {
+	poolName  string
+	k8sClient *m_k8s_client.K8sClient
+}
+
+func NewStoragePoolOperation(poolName string) (*storagePoolOperation, error) {
+	kc, err := m_k8s_client.NewK8sClient("")
+	if err != nil {
+		return nil, err
+	}
+	// Put pool object inside casPoolOperation object
+	return &storagePoolOperation{
+		poolName:  poolName,
+		k8sClient: kc,
+	}, err
+}
+
+func (s *storagePoolOperation) List() (*v1alpha1.StoragePoolList, error) {
+	if s.k8sClient == nil {
+		return nil, fmt.Errorf("Unable to fetch K8s client")
+	}
+
+	castName := menv.Get(menv.CASTemplateToListStoragePoolENVK)
+
+	// fetch read cas template specifications
+	cast, err := s.k8sClient.GetOEV1alpha1CAST(castName, mach_apis_meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	glog.Infof("%v", cast)
+	engine, err := engine.New(
+		cast,
+		"",
+		map[string]interface{}{},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := engine.Run()
+	if err != nil {
+		return nil, err
+	}
+
+	// unmarshall into openebs CasPool
+	sPool := &v1alpha1.StoragePoolList{}
+	err = json.Unmarshal(data, sPool)
+	if err != nil {
+		return nil, err
+	}
+	return sPool, nil
+}
+
+func (s *storagePoolOperation) Read() (*v1alpha1.StoragePool, error) {
+	if s.k8sClient == nil {
+		return nil, fmt.Errorf("Unable to fetch K8s client")
+	}
+
+	castName := menv.Get(menv.CASTemplateToReadStoragePoolENVK)
+
+	// fetch read cas template specifications
+	cast, err := s.k8sClient.GetOEV1alpha1CAST(castName, mach_apis_meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	engine, err := engine.New(
+		cast,
+		string(v1alpha1.StoragePoolTLP),
+		map[string]interface{}{
+			string(v1alpha1.OwnerCTP): s.poolName,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := engine.Run()
+	if err != nil {
+		return nil, err
+	}
+
+	glog.Infof("%s", data)
+
+	// unmarshall into openebs CasPool
+	sPool := &v1alpha1.StoragePool{}
```

Comment:
sPool---> sp
or 
sPool ----> storagePool
Let take view of  @AmitKumarDas . I am not sure.